tests/components/iaqualink/test_init.py

```diff
@@ -0,0 +1,148 @@
+"""Tests for iaqualink integration."""
+
+import asyncio
+from unittest.mock import AsyncMock, patch
+
+from iaqualink.device import (
+    AqualinkAuxToggle,
+    AqualinkBinarySensor,
+    AqualinkDevice,
+    AqualinkLightToggle,
+    AqualinkSensor,
+    AqualinkThermostat,
+)
+import iaqualink.exception
+from iaqualink.system import AqualinkSystem
+import pytest
+
+from homeassistant.components import iaqualink as ha_iaqualink
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.climate import DOMAIN as CLIMATE_DOMAIN
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.setup import async_setup_component
+
+from . import MOCK_CONFIG_DATA
+
+from tests.common import MockConfigEntry, mock_coro
+
+async_noop = AsyncMock(return_value=None)
+
+
+def _(cls, data=None):
+    """Create an aqualink class instance with little syntactic overhead."""
+    return cls(None, data if data else {})
+
+
+MOCK_CONFIG = MockConfigEntry(domain=ha_iaqualink.DOMAIN, data=MOCK_CONFIG_DATA)
+MOCK_SYSTEMS = {"SERIAL": _(AqualinkSystem)}
+MOCK_UNKNOWN_DEVICES = {"1": _(AqualinkDevice)}
+MOCK_DEVICES = {
+    "1": _(AqualinkAuxToggle),
+    "2": _(AqualinkBinarySensor),
+    "3": _(AqualinkLightToggle),
+    "4": _(AqualinkSensor),
+    "5": _(AqualinkThermostat),
+}
+
+
+async def test_no_config_creates_no_entry(hass):
+    """Test for when there is no iaqualink in config."""
+    with patch(
+        "homeassistant.components.iaqualink.async_setup_entry",
+        return_value=mock_coro(True),
```

Comment:
```suggestion
        return_value=True,
```
`mock_coro` is legacy for old versions of patch